build.gradle.kts

```diff
@@ -22,16 +22,15 @@ apply<GradleVersionsFilterPlugin>()
 apply<VersionCatalogUpdatePlugin>()
 apply<TaskTreePlugin>()
 
-// TODO (#983): use Gradle toolchain instead
-if (JavaVersion.current() != JavaVersion.VERSION_17) {
-  throw GradleException("Project must be built with JDK 17 (and not ${JavaVersion.current()}) since IntelliJ 2022.3 itself is compiled for Java 17")
-}
-
 fun getFlagsForAddOpens(vararg packages: String, module: String): List<String> {
   return packages.toList().map { "--add-opens=$module/$it=ALL-UNNAMED" }
 }
 
-val javaMajorVersion: JavaVersion by extra(JavaVersion.VERSION_11)
+// TODO (#859): bump to Java 17 once we no longer support InteliJ 2022.1 (the last version to run on Java 11)
+val targetJavaVersion: JavaVersion by extra(JavaVersion.VERSION_11)
```

Comment:
Class files of our plugin are & still will be compiled for Java 11 (i.e. classfile version 55.0).